test/master-playlist-controller.test.js

```diff
@@ -1184,16 +1192,103 @@ QUnit.test('correctly sets alternate audio track kinds', function(assert) {
                'spanish track\'s kind is "alternative"');
 });
 
+QUnit.test('trigger events when an AES is detected', function(assert) {
+  let hlsaes = 0;
+  let isAesCopy = Hls.Playlist.isAes;
+
+  Hls.Playlist.isAes = (media) => {
+    return true;
+  };
+
+  this.player.tech_.on('usage', (event) => {
+    if (event.name === 'hls-aes') {
+      hlsaes++;
+    }
+  });
+
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+
+  assert.equal(hlsaes, 1, 'an AES HLS stream is detected');
+  Hls.Playlist.isAes = isAesCopy;
+});
+
+QUnit.test('trigger events when an fMP4 stream is detected', function(assert) {
+  let hlsfmp4 = 0;
+  let isFmp4Copy = Hls.Playlist.isFmp4;
+
+  Hls.Playlist.isFmp4 = (media) => {
+    return true;
+  };
+
+  this.player.tech_.on('usage', (event) => {
+    if (event.name === 'hls-fmp4') {
+      hlsfmp4++;
+    }
+  });
+
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+
+  assert.equal(hlsfmp4, 1, 'an fMP4 stream is detected');
+  Hls.Playlist.isFmp4 = isFmp4Copy;
+});
+
+QUnit.test('trigger events when video and audio is demuxed by default', function(assert) {
+  let demuxed = 0;
+
+  this.player = createPlayer();
+  this.player.src({
+    src: 'manifest/multipleAudioGroups.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+
+  this.player.tech_.on('usage', (event) => {
+    if (event.name === 'hls-demuxed') {
+      demuxed++;
+    }
+  });
+
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+  // init segment
```

Comment:
the `hls-demuxed` event is triggered in `fillAudioTracks_` which is called after `loadedmetadata` handler which happens after the first non-master playlist request, so there should be no need to wait for a video or audio segment request to be fulfilled before seeing this event